src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/PremisesService.kt

```diff
@@ -91,4 +95,74 @@ class PremisesService(
 
       return success(lostBedsEntity)
     }
+
+  fun createNewPremises(requestBody: NewPremises) = validated<PremisesEntity> {
+
+    // TODO: dummy data for now - remove asap
+    val apAreaEntity = ApAreaEntity(
+      id = UUID.randomUUID(),
+      name = "arbitrary_ap_area",
+      identifier = "arbitrary_identifier",
+      probationRegions = mutableListOf()
+    )
+
+    val probationRegion = ProbationRegionEntity(
+      id = UUID.fromString("afee0696-8df3-4d9f-9d0c-268f17772e2c"), // Wales in db
+      name = "arbitrary_probation_region",
+      apArea = apAreaEntity,
+      premises = mutableListOf()
+    )
+    // end of dummy data
+
+    if (requestBody.address.isEmpty()) {
+      return generalError("Please provide an address")
```

Comment:
I think these should be field validation errors rather than general ones, e.g.

```
if (requestBody.address.isEmpty()) {
      "$.address" hasValidationError "Please provide an address"
}
```

then further down do a 
```
if (validationErrors.any()) {
      return fieldValidationError
    }
```

This way you'll get a more comprehensive output broken down by property rather than just the first issue.

Also on this - for the field validation stuff there's a mapping of error keys -> translations in our frontend somewhere.  I'm not super clear on how this actually works in the Node App, Stu or Rich should be able to provide more info if needed.  I would think your frontend will need to work the same way since you'll be using endpoints we've already built that return these keys (and we'll perhaps need to use this endpoint in the future too.)